x-pack/plugins/task_manager/server/task_store.ts

```diff
@@ -282,11 +270,15 @@ export class TaskStore {
           ),
           filterDownBy(InactiveTasks)
         ),
-        update: updateFields({
-          ownerId: this.taskManagerId,
-          status: 'claiming',
-          retryAt: claimOwnershipUntil,
-        }),
+        update: updateFieldsAndMarkAsFailed(
+          {
+            ownerId: this.taskManagerId,
+            retryAt: claimOwnershipUntil,
+          },
+          Object.entries(this.definitions).reduce((accumulator, [type, { maxAttempts }]) => {
+            return { ...accumulator, [type]: maxAttempts };
+          }, {})
+        ),
```

Comment:
Just a heads up - once the https://github.com/elastic/kibana/pull/80381 PR gets merged you'll have to change this as the DefinitionsDictionary becomes a more encapsulated entity (instead of `Object.entires` you should use the built in iterator).